firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/NotificationFeedbackTrigger.kt

```diff
@@ -0,0 +1,215 @@
+                    "Using a notification to initiate feedback to the developer. To enable this feature, allow the app post notifications."
+                )
+                .setPositiveButton("OK") { _, _ ->
+                    Log.i(TAG, "Launching request for permission.")
+                    requestPermissionLauncher!!.launch(POST_NOTIFICATIONS)
+                }
+                .show()
+        } else {
+            Log.i(TAG, "Launching request for permission without rationale.")
+            requestPermissionLauncher!!.launch(POST_NOTIFICATIONS)
+        }
+    }
+
+    private fun cancelNotification(context: Context) {
+        val notificationManager = NotificationManagerCompat.from(context)
+        Log.i(TAG, "Cancelling notification")
+        notificationManager.cancel(FEEDBACK_NOTIFICATION_ID)
+    }
+
+    override fun onActivityResumed(activity: Activity) {
+        if (isEnabled) {
+            if (activity !is TakeScreenshotAndTriggerFeedbackActivity) {
+                Log.d(TAG, "setting current activity")
+                currentActivity = activity
+            }
+            showNotification(activity)
+        }
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        cancelNotification(activity)
+    }
+
+    override fun onActivityDestroyed(activity: Activity) {
+        Log.d(TAG, "clearing current activity")
+        currentActivity = null
+    }
+
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        if (activity is ActivityResultCaller) {
+            requestPermissionLauncher =
+                activity.registerForActivityResult(ActivityResultContracts.RequestPermission()) {
+                        isGranted: Boolean ->
+                    if (!isEnabled) {
+                        Log.w(TAG, "Trigger disabled after permission check. Abandoning notification.")
+                    } else if (isGranted) {
+                        showNotification(activity)
+                    } else {
+                        Log.i(TAG, "Permission not granted")
+                        // TODO: Ideally we would show a message indicating the impact of not
+                        //   enabling the permission, but there's no way to know if they've
+                        //   permanently denied the permission, and we don't want to show them a
+                        //   message after each time we try to post a notification.
+                    }
+                }
+        } else {
+            Log.w(TAG, "Not listening for screenshots because this activity can't register for permission request results: $activity")
+        }
+    }
+
+    // Other lifecycle methods
+    override fun onActivityStarted(activity: Activity) {}
+    override fun onActivityStopped(activity: Activity) {}
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
+
+    fun takeScreenshot() {
+        val activity = currentActivity
```

Comment:
So it looks like you are able to take a screenshot of a paused activity then, very cool. Out of curiosity, did you try taking a screenshot of the `TakeScreenshotAndTriggerFeedbackActivity` itself, and if so what happened?

Reply:
It just took a screenshot of the empty activity.